### ISSUE TITLE
The vec2 exercise is inappropriate

### ISSUE THREAD

**Issue by senekor** [rust-lang/rustlings#2102]
The following commit added significant complexity to the exercise: https://github.com/rust-lang/rustlings/commit/8e1f617d3402cb05b05c6737f60fbbfe74da4d78

I understand the desire to show people functional programming, I love it as much as the next Rust developer. But this exercise comes pretty much right after _primitive types_.

Even using vectors is a stretch here, the book itself covers vectors only much later. But I think that's withing reason here, because vectors are necessary to do almost anything and most programmers will be familiar with something similar. So I think it's justified to cover vectors a bit earlier than the book.

However... iterators? map? collect? lambda syntax? I find this completely unjustified. If this content is not covered in the dedicated section for iterators, it can be moved there.

Please consider removing the `vec_map` part from that exercise so as not to overwhelm new learners too early.

**Comment by mo8it** [comment]
I agree. I will remove it from the exercise in v7. But I will keep it in the solution as one of the possible solutions. I will mention that we will deal with iterators later.

**Comment by jgarte** [comment]
Hi, what's the status on this one? Thanks for working on this.
